swim.go

```diff
@@ -130,27 +139,43 @@ func (n *Node) runReceive() {
 		if err := json.Unmarshal(b[:len], &e); err != nil {
 			continue
 		}
-		e.P.remoteID = e.FromID
+		e.P.remoteID = e.SrcID
 		e.P.remoteAddr = addr
-		ps, us, ok := n.receive(e.P)
+		ps, ok := n.receive(e.P)
 		if !ok {
 			return
 		}
-		n.sendUpdates(us)
+		n.emitPending()
 		n.send(ps)
 	}
 }
 
-func (n *Node) receive(p packet) ([]packet, []Update, bool) {
+func (n *Node) receive(p packet) ([]packet, bool) {
 	n.mu.Lock()
 	defer n.mu.Unlock()
 	return n.fsm.receive(p)
 }
 
-func (n *Node) sendUpdates(us []Update) {
-	for _, u := range us {
+// emitPending sends all pending Updates and Messages on their respective
+// channels.
+func (n *Node) emitPending() {
+	n.mu.Lock()
+	defer n.mu.Unlock()
+	for _, u := range n.fsm.pendingUpdates {
 		n.updates.Send() <- u
 	}
+	n.fsm.pendingUpdates = nil
+	for _, m := range n.fsm.pendingMessages {
+		n.messages.Send() <- m
+	}
+	n.fsm.pendingMessages = nil
+}
+
+// Message disseminates b throughout the network.
+func (n *Node) Message(b []byte) {
```

Comment:
Is this method name using the word "Message" as a verb? That seems a little too precious imo, especially because we already have both `message` and `Message` elsewhere as nouns.